Added missing API docs sidebar links

zmaglica opened the pull request:
#### Changes proposed in this Pull Request
Added missing API docs link in sidebar for https://github.com/Automattic/woocommerce-payments/pull/7384 and https://github.com/Automattic/woocommerce-payments/pull/7369

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Make sure to see links for payment methods and payment intent.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

zmaglica committed:
Added documentation for create payment intent

zmaglica committed:
Merge branch 'develop' into fix/docs-api-add-missing-sidebar-links-for-intent-and-pm

zmaglica committed:
Added missing sidebar links

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7389` or branch name `fix/docs-api-add-missing-sidebar-links-for-intent-and-pm` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/docs-api-add-missing-sidebar-links-for-intent-and-pm) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 25012d7b4c10b5019ed7c97bdeb635f8325472cb
- Build time: 2023-10-05 13:28:49 UTC

_Note: the build is updated when a new commit is pushed to this PR._

RadoslavGeorgiev reviewed (approved):
LGTM 🚢 

Please add a changelog entry, though. You can select `patch` -> `dev` to avoid shipping it, but there should be a file.

<img width="1581" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/5311119/25b5ae9d-de4a-428a-8d89-f79ba0ec5b24">

zmaglica committed:
Fixed typo

zmaglica committed:
Added changelog

zmaglica commented:
> LGTM 🚢
> 
> Please add a changelog entry, though. You can select `patch` -> `dev` to avoid shipping it, but there should be a file.
> 
> <img alt="image" width="1581" src="https://user-images.githubusercontent.com/5311119/272913723-25b5ae9d-de4a-428a-8d89-f79ba0ec5b24.png">

I have added a changelog and fixed a small typo.
